code/modules/antagonists/clockcult/clock_items/clock_weapons/ratvarian_shield.dm

```diff
@@ -0,0 +1,84 @@
+
+/obj/item/shield/riot/ratvarian/examine(mob/user)
+	if((is_servant_of_ratvar(user) || isobserver(user)))
+		desc = clockwork_desc
+		desc +="\n <span class='inathneq_small'>The shield has absorbed [dam_absorbed] damage, multiplying the effectiveness of its bashes by [calc_bash_mult()]</span>"
+	. = ..()
+	desc = initial(desc)
+
+obj/item/shield/riot/ratvarian/proc/calc_bash_mult()
+	var/bash_mult = 0
+	if(!dam_absorbed)
+		return 1
+	else
+		bash_mult += round(clamp(1 + (dam_absorbed / bash_mult_steps), 1, max_bash_mult), 0.1) //Multiplies the effect of bashes by up to [max_bash_mult], though never less than one
+		return bash_mult
+
+/obj/item/shield/riot/ratvarian/proc/calc_bash_absorb_use()
+	var/absorb_use = 0
+	absorb_use = max(0, round(dam_absorbed * (calc_bash_mult() / round(1 + (dam_absorbed / bash_mult_steps), 0.1)), 1)) //Calculates how much of the absorbed damage the bash would actually use, so its not wasted
+	return absorb_use
+
+/obj/item/shield/riot/ratvarian/on_shield_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
+	if(!damage)
+		return ..()
+
+	if(!is_servant_of_ratvar(owner))
+		owner.visible_message("<span class='warning'>As [owner] blocks the attack with [src], [owner.p_they()] suddenly drops it, whincing in pain! </span>", "<span class='warning'>As you block the attack with [src], it heats up tremendously, forcing you to drop it from the pain alone! </span>")
+		owner.emote("scream")
+		playsound(src, 'sound/machines/fryer/deep_fryer_emerge.ogg', 50)
+		if(iscarbon(owner)) //Type safety for if a drone somehow got a shield (ratvar protect us)
+			var/mob/living/carbon/C = owner
+			var/obj/item/bodypart/part = C.get_holding_bodypart_of_item(src)
+			C.apply_damage((iscultist(C) ? damage * 2 : damage), BURN, (istype(part, /obj/item/bodypart/l_arm) ? BODY_ZONE_L_ARM : BODY_ZONE_R_ARM)) //Deals the damage to the holder instead of absorbing it instead + forcedrops. Doubled if a cultist of Nar'Sie.
+		else
+			owner.adjustFireLoss(iscultist(owner) ? damage * 2 : damage)
+		addtimer(CALLBACK(owner, /mob/living.proc/dropItemToGround, src, TRUE), 1)
+	else if(!is_servant_of_ratvar(attacker)) //No exploiting my snowflake mechanics
+		dam_absorbed += damage
+		playsound(owner,  'sound/machines/clockcult/steam_whoosh.ogg', 30)
+
+	if(damage <= 10) //The shield itself is hard to break, this DOES NOT modify the actual blocking-mechanic
+		damage = 0
+	else
+		damage -= 5
+	return ..()
+
+/obj/item/shield/riot/ratvarian/shatter(mob/living/carbon/human/owner)
+	playsound(owner, 'sound/magic/clockwork/anima_fragment_death.ogg', 50)
+	new /obj/item/clockwork/alloy_shards/large(get_turf(src))
+
+/obj/item/shield/riot/ratvarian/user_shieldbash(mob/living/user, atom/target, harmful)
+	if(harmful && is_servant_of_ratvar(user)) // No fun for non-clockies, but you can keep the normal bashes. Until you try to block with it.
```

Comment:
I suggest an early return if it is either non-harmful or the user isn't a clockie instead.

Reply:
Did exactly that.